.github/workflows/ci.yml

```
@@ -1,56 +1,87 @@
     steps:
       - name: Checkout branch
         uses: actions/checkout@v4
         with:
           ref: ${{ github.head_ref }}
       - name: Fetch full git history
         run: git fetch --prune --unshallow
-      - name: Set up Micromamba env
-        uses: mamba-org/setup-micromamba@422500192359a097648154e8db4e39bdb6c6eed7
+      - name: Set up pixi
+        uses: prefix-dev/setup-pixi@19b5e6071f70a82d3c5741d7b3f98a71d192eb20
         with:
-          environment-file: ${{ matrix.environment-file }}
-          create-args: >-
-            pytest-emoji
-            pytest-md
-      - name: pip install
-        run: pip install -e .
+          environments: ${{ matrix.environment }}
+      - name: Install repository
+        run: pixi run -e ${{ matrix.environment }} postinstall
       - name: Run pytest
-        uses: pavelzw/pytest-action@v2
+        id: pytest
+        run: |
+          REPORT="${{ runner.temp }}/report"
+          echo "report-path=$REPORT" >> $GITHUB_OUTPUT
+          pixi run -e ${{ matrix.environment }} pytest -v --md=$REPORT --emoji
+        shell: bash -el {0} # use login shell to source .bashrc for environments
+      - name: Add report to summary
+        run: |
+          echo "::group::..."
+          FINAL_REPORT="${{ runner.temp }}/output"
+          echo "# Unit tests ${{ matrix.environment }}" > "$FINAL_REPORT"
+          echo "<details><summary>Click to expand!</summary>" >> "$FINAL_REPORT"
+          tail -n+2 "${{ steps.pytest.outputs.report-path }}" >> "$FINAL_REPORT"
+          echo "</details>" >> "$FINAL_REPORT"
+          cat "$FINAL_REPORT" >> "$GITHUB_STEP_SUMMARY"
+          echo "::endgroup::"
+          echo
+          echo "====================================================================================="
+          echo Markdown summaries: "$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID"
+          echo "====================================================================================="
+        shell: bash
+
+  mypy-type-checks:
+    name: Mypy Type Checks
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+      - name: Set up pixi
+        uses: prefix-dev/setup-pixi@632d17935141ec801697e2c359784b878adecbbe
+      - name: Install repository
+        run: pixi run postinstall
+      - name: Run mypy
+        run: pixi run mypy
 
   pre-commit-checks:
-    name: Pre-commit checks - Python 3.12
+    name: Pre-commit Checks
     timeout-minutes: 30
     runs-on: ubuntu-latest
     steps:
       - name: Checkout branch
         uses: actions/checkout@v4
-      - name: Run pre-commit-conda
-        uses: quantco/pre-commit-conda@v1
+      - name: Set up pixi
+        uses: prefix-dev/setup-pixi@d72307a91ccc8a856376cdb9ff5b074a04422409
         with:
-          python-version: "3.12"
+          environments: lint
```

```suggestion
          environments: lint default
```

when adding mypy as a pre-commit hook